pkg/retry/options.go

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package retry
+
+import "time"
+
+// Options is the functional options struct.
+type Options struct {
+	Units  time.Duration
+	Jitter time.Duration
+}
+
+// Option is the functional option func.
+type Option func(*Options)
+
+// WithUnits is a functional option for setting the units of the ticker.
+func WithUnits(o time.Duration) Option {
+	return func(args *Options) {
+		args.Units = o
+	}
+}
+
+// WithJitter is a functional option for setting the jitter flag.
+func WithJitter(o time.Duration) Option {
+	return func(args *Options) {
+		args.Jitter = o
+	}
+}
+
```

Comment:
I feel like we should expose a max interval setting ( specifically for exponential backoff ). Ex, I want to wait a total 10min with a max of 30s intervals and exponentially grow from 1s.

Reply:
Potentially. Certainly something we can improve on.